thau-api/src/main/java/com/mgrin/thau/sessions/SessionsAPI.java

```
@@ -0,0 +1,125 @@
+
+    private CredentialsService credentials;
+
+    private ThauConfigurations configurations;
+
+    @Autowired
+    public SessionsAPI(SessionsService sessions, UsersService users, CredentialsService credentials, ThauConfigurations configurations) {
+        this.sessions = sessions;
+        this.users = users;
+        this.credentials = credentials;
+        this.configurations = configurations;
+    }
+    @GetMapping(produces = { "application/json" })
+    public Session getCurrentSessoin(@RequestHeader(name = SessionsAPI.JWT_HEADER, required = false) String token) {
+        try {
+            Session session = sessions.getSessionFromToken(token);
+            if (session.getStrategy() == Strategy.PASSWORD) {
+                Credentials creds = credentials.getByEmail(session.getUser().getEmail());
+                if (creds == null || (configurations.getPasswordStrategyConfiguration().isRequireEmailVerification() && !creds.isVerified())) {
+                    throw new APIError(HttpStatus.FORBIDDEN, "Email is not yet validated");
+                }
+            }
+            return session;
+        } catch (APIError error) {
+            throw error;
+        } catch (Exception exception) {
+            throw new APIError(HttpStatus.UNAUTHORIZED, "Unauthorized", exception);
+        }
+    }
+
+    @PostMapping(path = "/password", consumes = {"application/json"}, produces = {"application/json"})
+    public TokenDTO createSession(@RequestBody PasswordAuthDTO auth, HttpServletResponse response) {
+        try {
+            Credentials creds = credentials.getByEmailAndPassword(auth.getEmail(), auth.getPassword());
+            Session session = sessions.create(creds.getUser(), Strategy.PASSWORD);
+            String token = sessions.createJWTToken(session);
+            response.setHeader(JWT_HEADER, token);
+            return new TokenDTO(token);
+        } catch (EntityNotFoundException e) {
+            throw new APIError(HttpStatus.NOT_FOUND, e);
```

This is what I meant in that earlier comment with the @ResponseStatus that you can provide - you can throw an Exception with a message, and the status code is already provided with it. No need for bulky try-catch blocks as you use them here.